app/src/org/odk/collect/android/widgets/AudioWidget.java

```diff
@@ -70,10 +71,9 @@ public void onClick(View v) {
                 Intent i = new Intent(android.provider.MediaStore.Audio.Media.RECORD_SOUND_ACTION);
                 i.putExtra(android.provider.MediaStore.EXTRA_OUTPUT,
                         android.provider.MediaStore.Audio.Media.EXTERNAL_CONTENT_URI.toString());
-                mWaitingForData = true;
                 try {
-                    ((Activity)getContext())
-                            .startActivityForResult(i, FormEntryActivity.AUDIO_VIDEO_FETCH);
+                    ((Activity)getContext()).startActivityForResult(i, FormEntryActivity.AUDIO_VIDEO_FETCH);
+                    pendingCalloutInterface.setPendingCalloutFormIndex(prompt.getIndex());
```

Comment:
I don't think it actually matters, but I usually try to make any calls to a startActivity() method the last call I make in a sequence.